lib/routes/auth.js

```diff
@@ -44,4 +48,43 @@ router.post(
   }
 );
 
+// /api/auth/login
+router.post(
+  './login',
+  [
+    check('email', 'This email address is not registered')
+      .normalizeEmail()
+      .isEmail(),
+    check('password', 'Invalid password').exists(),
+  ],
+  async (request, response) => {
+    try {
+      const errors = validationResult(request);
+      if (!errors.isEmpty) {
+        return response.status(400).json({
+          errors: errors.array(),
+          message: 'Invalid log in data',
+        });
+      }
+
+      const { email, password } = request.body;
+      const user = await User.findOne({ email: email });
+      if (!user) {
+        return response.status(400).json({ message: 'User not found' });
+      }
+
+      const passwordCheck = await bcrypt.compare(password, user.password);
+      if (!passwordCheck) {
+        return response
+          .status(400)
+          .json({ message: 'Incorrect username or password' });
+      }
```

Comment:
What happens when the `passwordCheck` is true?
You won't have any response here -> endless request.

Reply:
If it´s not the case, in would response (88) `response.json({ token, userId: user.id })`
Or wouldn't it work?